### DISCUSSION TITLE
JSONPath help -> $.*~ and $.*^

### DISCUSSION THREAD

**Discussion by ssaravanan2** [duckdb/duckdb#10759] [Q&A] [answered]
What is the equivalent to get JSONPath [$.*~ ] functionality ?

Example:
{
    "firstName": "John",
    "lastName": "doe",
    "age": 26,
    "address": {
        "streetAddress": "naist street",
        "city": "Nara",
        "postalCode": "630-0192"
    },
    "phoneNumbers": [
        {
            "type": "iPhone",
            "number": "0123-4567-8888"
        },
        {
            "type": "home",
            "number": "0123-4567-8910"
        }
    ]
}

JSONPath $.*~ will fetch...
[
  "firstName",
  "lastName",
  "age",
  "address",
  "phoneNumbers"
]

**Comment by cmdlineluser** [comment, marked as answer]
That looks like `json_keys()` ?

- https://duckdb.org/docs/extensions/json#json-scalar-functions

**Reply by ssaravanan2** [reply]
Thanks -that is it.

This worked -> select json_keys(j)  from example
